src/main/java/layron/tms/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+import layron.tms.model.Role;
+import layron.tms.model.RoleName;
+import layron.tms.model.User;
+import layron.tms.repository.RoleRepository;
+import layron.tms.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final RoleRepository roleRepository;
+    private final SecurityConfig config;
+
+    @Override
+    public UserRegistrationResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("User already exists!");
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(config.getPasswordEncoder().encode(requestDto.getPassword()));
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        Role role = roleRepository.findByName(RoleName.ROLE_USER)
+                .orElseThrow(() -> new RegistrationException("There is no role for this user!"));
+        user.setRoles(Set.of(role));
+        User savedUser = userRepository.save(user);
+        return userMapper.toUserResponse(savedUser);
+    }
+
+    //TODO: Is there any logic to use this exception here?
+    @Override
+    public UserDto getUser() throws UserNotFoundException {
+        Authentication loggedInUser = SecurityContextHolder.getContext().getAuthentication();
+        String username = loggedInUser.getName();
+        User user = userRepository.findByUsername(username)
+                .orElseThrow(() ->
+                        new UserNotFoundException("There is no user with such username!"));
+        return userMapper.toDto(user);
+    }
+
+    @Override
+    public UserDto updateUser(
+            UpdateUserRequestDto requestDto
+    ) throws UserNotFoundException {
+        Authentication loggedInUser = SecurityContextHolder.getContext().getAuthentication();
+        User user = userRepository.findByUsername(loggedInUser.getName())
+                .orElseThrow(() ->
+                        new UserNotFoundException("There is no user with such username!"));
+        if (requestDto.email() != null && !requestDto.email().isEmpty()) {
+            user.setEmail(requestDto.email());
+        }
+        if (requestDto.firstName() != null && !requestDto.firstName().isEmpty()) {
+            user.setFirstName(requestDto.firstName());
+        }
+        if (requestDto.lastName() != null && !requestDto.lastName().isEmpty()) {
+            user.setLastName(requestDto.lastName());
+        }

```

Comment:
most probably you can utilize following method to make code shorter
import static liquibase.util.StringUtil.isNotEmpty;